Ansible 4 roadmap

abadger opened the pull request:
##### SUMMARY
Schedule for the release of Ansible-4.x

This is going to be discussed and approved at an Ansible Community IRC Meeting.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

abadger committed:
Ansible-4 roadmap

Copy the ansible-3.x roadmap over for a somewhat helpful diff

abadger committed:
Schedule for ansible-4.0

dmsimard reviewed (approved):
Thanks for putting this together!

Looks good to me -- it even includes a link to the 2.11 roadmap.

gundalow commented on docs/docsite/rst/roadmap/COLLECTIONS_4.rst:
```diff
@@ -0,0 +1,49 @@
+
+
+:2021-01-26: New Collections will be reviewed for inclusion in ansible-4.
+:2021-03-03: ansible-4.0.0 alpha1 (biweekly alpha releases.  This is timed to coincide with the start of the ansible-core-2.11 pre-releases)
```
```suggestion
:2021-03-03: ansible-4.0.0 alpha1 (The ``ansible`` alphas are timed to coincide with the start of the ``ansible-core-2.11`` pre-releases)
```

abadger replied:
I'd like to keep biweekly as all the other pre-releases are released weekly.  Maybe:

```suggestion
:2021-03-03: ansible-4.0.0 alpha1 (biweekly ``ansible`` alphas.  These are timed to coincide with the start of the ``ansible-core-2.11`` pre-releases)
```

gundalow replied:
Sounds good

gundalow committed:
biweekly releases

Co-authored-by: Toshio Kuratomi <a.badger@gmail.com>

dericcrago commented on docs/docsite/rst/roadmap/COLLECTIONS_4.rst:
```diff
@@ -0,0 +1,49 @@
+
+
+:2021-01-26: New Collections will be reviewed for inclusion in ansible-4.
+:2021-03-03: ansible-4.0.0 alpha1 (biweekly ``ansible`` alphas.  These are timed to coincide with the start of the ``ansible-core-2.11`` pre-releases)
+:2021-03-17: ansible-4.0.0 alpha2
```
were these two dates intentionally on Wednesday or should they be shifted to Tuesday?

```suggestion
:2021-03-02: ansible-4.0.0 alpha1 (biweekly ``ansible`` alphas.  These are timed to coincide with the start of the ``ansible-core-2.11`` pre-releases)
:2021-03-16: ansible-4.0.0 alpha2
```

abadger replied:
Yes... It's easier to see when I have the ansible-core pre-releases interspersed: https://hackmd.io/y7BBcweNR3aRVLuMbKkDxw#Proposed-Schedule-for-Ansible-40   The idea is for the ansible alphas to release the day after the ansible-core betas and release candidcates.

abadger replied:
Okay, relrod has finalized the dates for alpha2 and so forth: https://github.com/ansible/ansible/pull/73560/files

ansible-core-2.11-beta1 will still be on a Tuesday but subsequent releases will be shifted to Monday.  So I'll change our ansible-4.0.0 alpha2 to be a Tuesday.

About why beta1 is on a Tuesday, it is a combination of publishing that date quite a few months ago so it seems too late to change and it gives people one extra day to get their work finished before the beta freeze.

dericcrago commented on docs/docsite/rst/roadmap/COLLECTIONS_4.rst:
```diff
@@ -0,0 +1,49 @@
+:2021-04-21: Community IRC Meeting topic: Decide what contingencies to activate for any blockers that do not meet the deadline.
+:2021-04-26: last day for new collections to be **reviewed and approved** for inclusion in ansible-4
+:2021-04-26: last day for collections to make backwards incompatible releases that will be accepted into Ansible-4.
+:2021-04-27: ansible-4.0.0 beta1 -- feature freeze [1]_ (weekly beta releases.  Collection owners and interested users
+             should test for bugs)
```
extra whitespace?

```suggestion
:2021-04-27: ansible-4.0.0 beta1 -- feature freeze [1]_ (weekly beta releases.  Collection owners and interested users should test for bugs)
```

abadger replied:
I cut lines at 100 characters for easy reading as text.  But rst will parse either one correctly when transforming to html.

abadger committed:
Update docs/docsite/rst/roadmap/COLLECTIONS_4.rst

Co-authored-by: Deric Crago <deric.crago@gmail.com>

abadger committed:
The ansible-core beta2 release date has changed.

So move our alpha2 date to one day after their beta2

https://github.com/ansible/ansible/pull/73560/files

acozine committed:
fix capitalization and punctuation

acozine reviewed (approved):
Looks good.

samccann commented on docs/docsite/rst/roadmap/COLLECTIONS_4.rst:
```diff
@@ -0,0 +1,48 @@
+:2021-03-30: Ansible-4.0.0 alpha3
+:2021-04-13: Last day for new collections to be submitted for inclusion in Ansible 4. Note that collections MUST be reviewed and approved before being included. There is no guarantee that we will review every collection. The earlier your collection is submitted, the more likely it will be that your collection will be reviewed and the necessary feedback can be addressed in time for inclusion.
+:2021-04-13: Ansible-4.0.0 alpha4
+:2021-04-14: Community IRC Meeting topic: list any new collection reviews which block release.  list any backwards incompatible collection releases that beta1 should try to accommodate.
```
```suggestion
:2021-04-14: Community IRC Meeting topic: list any new collection reviews which block release.  List any backwards incompatible collection releases that beta1 should try to accommodate.
```

acozine committed:
Update docs/docsite/rst/roadmap/COLLECTIONS_4.rst

Co-authored-by: Sandra McCann <samccann@redhat.com>
